R/spread.R

```diff
@@ -94,14 +102,23 @@ spread_.data.frame <- function(data, key_col, value_col, fill = NA,
   if (!is.na(fill)) {
     ordered[is.na(ordered)] <- fill
   }
-  if (convert) {
-    ordered <- type.convert(ordered, as.is = TRUE)
-  }
 
+  if (convert && !is.character(ordered)) {
+    ordered <- as.character(ordered)
+  }
   dim(ordered) <- c(attr(row_id, "n"), attr(col_id, "n"))
   colnames(ordered) <- as.character(col_labels[[1]])
+  ordered <- matrixToDataFrame(ordered)
+
+  if (convert) {
+    ordered[] <- lapply(ordered, type.convert, as.is = TRUE)
+  }
+
+  if (is.factor(ordered[[1]]) && drop) {
```

Comment:
I think this is a different meaning of drop compared to the documentation, so I don't think it's necessary